ci/rpm_scan_daos_test.sh

```diff
@@ -5,19 +5,22 @@ set -uex
 # DAOS_PKG_VERSION environment variable needs to be set for this script
 
 lmd_tarball='maldetect-current.tar.gz'
-# todo: Move to tools repository
-curl "http://rfxn.com/downloads/${lmd_tarball}" \
-  -z ${lmd_tarball}  --silent --show-error --fail -O
 
-nodelist=("${NODELIST//,/ }")
-scp -i ci_key "${lmd_tarball}" "jenkins@${nodelist[0]}:/var/tmp"
+lmd_url="${JENKINS_URL}job/daos-stack/job/tools/job/master/"
+lmd_url+="lastSuccessfulBuild/artifact/${lmd_tarball}"
+curl "${lmd_url}" -z "./${lmd_tarball}"  --silent --show-error --fail -O
+
+nodelist=(${NODELIST//,/ })
 
 mydir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
-# shellcheck disable=SC2029
-ssh "$SSH_KEY_ARGS" jenkins@"$NODE" "NODE=${nodelist[0]}  \
-  DAOS_PKG_VERSION=$DAOS_PKG_VERSION                      \
-  $(cat "$mydir/rpm_scan_daos_test_node.sh")"
+scp -i ci_key "${lmd_tarball}" "$mydir/rpm_scan_daos_test_node.sh" \
+              "jenkins@${nodelist[0]}:/var/tmp"
+
+ssh "$SSH_KEY_ARGS" jenkins@"${nodelist[0]}" \
+ "NODE=${nodelist[0]}                \
+  DAOS_PKG_VERSION=$DAOS_PKG_VERSION \
+  /var/tmp/rpm_scan_daos_test_node.sh"
```

Comment:
Why break this up into a copy-then-execute?  The downside of it is that it's introducing one more network round-trip that adds delay (ssh handshakes are not cheap) and potential for environmental (i.e. network, disk, etc.) failures.

What happens, for example, if `/var/tmp/` is full?  I think if you want to go this route, you need to error check that and verify that the script you copied made it there and is vaild (i.e. not empty -- or even just incomplete -- because of a (nearly) full disk for example).  Executing it inline has none of these issues.

Reply:
Ashley requested doing it this way as in the previous comment.

Can you two come to an agreement on which way to do this?

Reply:
I much prefer this way, as it avoids any potential escaping issues and I've really struggled with that with the in-line scripts in the Jenkinsfile, but this is more a question for @brianjmurrell than me, so I'm happy to bow to his judgement.

Reply:
The cool part is that as `cat`ted inline scripts, they don't need to be escaped:

```
$ pwd
/home/brian/daos/daos
$ ssh root@wolf-110 "echo $PWD"
/home/brian/daos/daos  # PWD was resolved locally before sending the script to the remote
# now the exact same script in a file and cat-ted "inline"
$ cat /tmp/script.sh
echo $PWD
$ ssh root@wolf-110 "$(cat /tmp/script.sh)"
/root
```

As you can see, the `$PWD` in the script didn't need escaping and was not resolved on the local side before being sent as an inline script to the remote.

Reply:
Changed back to use catted scripts.